src/core/jsonSchema.ts

```diff
@@ -23,6 +23,8 @@ export interface NormalizedSchema extends Schema {
 }
 
 export function parseSchema(content: any, url?: string): Schema {
+    content = JSON.parse(JSON.stringify(content));
```

Comment:
I'm not sure I understand, but. this will prevent sharing of values inside the object which seem to cause dtsgenerator to fail. As an specific example this happens when js-yaml is used to parse a yaml file that contains yaml references.

If you are asking why do this here and not closer to the call to js-yaml then I'd argue that requiring that the input object does not have shared values is easy to for any callers of the constructor, not just for stuff that comes from js-yaml, to get wrong.

Reply:
i see